Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -0,0 +1,132 @@
+import UIKit
+import PayPalNativeCheckout
+import PayPalCheckout
+import PaymentsCore
+import Card
+
+class PayPalViewModel: ObservableObject, PayPalDelegate {
+
+    enum State {
+        case initial
+        case loading(content: String)
+        case mainContent(title: String, content: String, flowComplete: Bool)
+    }
+
+    @Published private(set) var state = State.initial
+
+    private var accessToken = ""
+    private var payPalClient: PayPalClient?
+
+    func getAccessToken() {
+        state = .loading(content: "Getting access token")
+        Task {
+            guard let token = await getAccessToken() else {
+                publishStateToMainThread(.mainContent(title: "Access Token", content: accessToken, flowComplete: false))
+                return
+            }
+            accessToken = token
+            payPalClient = PayPalClient(config: CoreConfig(accessToken: token, environment: PaymentsCore.Environment.sandbox))
+            payPalClient?.delegate = self
+            publishStateToMainThread(.mainContent(title: "Access Token", content: accessToken, flowComplete: false))
+        }
+    }
+
+    func retry() {
+        payPalClient = nil
+        accessToken = ""
+        state = .initial
+    }
+
+    func checkoutWithOrderID() {
+        state = .loading(content: "Initializing checkout")
+        Task {
+            do {
+                let orderID = try await self.getOrderID()
+                await self.payPalClient?.start { createOrderAction in
+                    createOrderAction.set(orderId: orderID)
+                }
+            } catch let error {
+                publishStateToMainThread(.mainContent(title: "Error", content: "\(error.localizedDescription)", flowComplete: true))
+            }
+        }
+    }
+
+    private func getOrderID() async throws -> String {
+        let order = try await DemoMerchantAPI.sharedService.createOrder(
+            orderRequest: OrderRequestHelpers.getOrderRequest()
+        )
+        return order.id
+    }
+
+    func getAccessToken() async -> String? {
+        await DemoMerchantAPI.sharedService.getAccessToken(environment: DemoSettings.environment)
+    }
+
+    // MARK: - PayPalDelegate conformance
```

Comment:
In our [Swift Style Guide draft](https://hackmd.io/kIyOnIFIRrKY4b5wQ0dUpw?both#Protocol-Conformance) we call out that protocol conformances should be moved into extensions